nmpolicy/internal/expander/stateExpander.go

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 NMPolicy Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package expander
+
+import (
+	"fmt"
+	"regexp"
+
+	yaml "sigs.k8s.io/yaml"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/capture"
+	"github.com/nmstate/nmpolicy/nmpolicy/types"
+)
+
+type StateExpander struct {
+	capResolver capture.PathCapturer
+}
+
+func New(capResolver capture.PathCapturer) StateExpander {
+	return StateExpander{capResolver: capResolver}
+}
+
+func (c StateExpander) Expand(captures map[string]types.CaptureState, rawDesrideState []byte) ([]byte, error) {
+	desiredState, err := unmarshalState(rawDesrideState)
+	if err != nil {
+		return nil, fmt.Errorf("expand error: %v", err)
+	}
+	expandedState, err := c.expandState(captures, desiredState)
+	if err != nil {
+		return nil, fmt.Errorf("expand error: %v", err)
+	}
+
+	rawExpandedState, err := marshalState(expandedState)
+	if err != nil {
+		return nil, fmt.Errorf("expand error: %v", err)
+	}
+	return rawExpandedState, nil
+}
+
+func unmarshalState(desrideState []byte) (interface{}, error) {
+	var unmarshaledState interface{}
+	err := yaml.Unmarshal(desrideState, &unmarshaledState)
+	if err != nil {
+		return nil, fmt.Errorf("failed unmarshaling the state %v : %v", desrideState, err)
+	}
+
+	return unmarshaledState, nil
+}
+
+func marshalState(state interface{}) ([]byte, error) {
+	rawState, err := yaml.Marshal(state)
+	if err != nil {
+		return nil, fmt.Errorf("failed marshaling the expanded state %v : %v", state, err)
+	}
+
+	return rawState, nil
+}
```

Comment:
Except for the error wrapping I don't see much use for this function, we are not encapsulating much logic.

Reply:
removed the method